x/erc20/keeper/ibc_callbacks.go

```diff
@@ -89,6 +89,12 @@ func (k Keeper) OnRecvPacket(
 		data.Denom, data.Amount,
 	)
 
+	// If the coin denom starts with `factory/` then it is a token factory coin, and we should not convert it
+	// NOTE: Check https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/ for more information
+	if strings.HasPrefix(data.Denom, "factory/") {
```

Comment:
there is another form `factory:*` (check cosmos chain-registry) so I suggest the statement should be:
`starts with "factory" /*performance reason*/ && regex factory[^a-zA-Z\d]`

Reply:
<img width="705" alt="Screen Shot 2024-02-15 at 18 14 53" src="https://github.com/evmos/evmos/assets/107764609/27188842-1c24-4a5f-9566-3a74fe768b4d">

Reply:
Hey @VictorTrustyDev where did you spot this pattern, I can't find any reference to these denoms in the chain registry? Could it be that they are being modified for display purposes only in the query?

Reply:
@Vvaradinov this one in [Osmosis assets record on chain-registry](https://github.com/cosmos/chain-registry/blob/d9d923b0b92aabd875b0b0176ca07c3a409a8574/osmosis/assetlist.json#L5887) is the prove of the format

Reply:
Yes it seems that it's just a visualization issue because their tokenfactory should be a fork of Osmosis:

- https://github.com/Team-Kujira/core/blob/b8ab1241c8763329ba3a228fb1a1de69104467ef/x/denom/spec/README.md#L1-L1

- The denom is deconstructed using `/`: https://github.com/Team-Kujira/core/blob/b8ab1241c8763329ba3a228fb1a1de69104467ef/x/denom/types/denoms.go#L30-L57

But I'm not 100% since the repo is not fully opensource.

Reply:
@VictorTrustyDev seems like one of them is the path while we explicitly check for the actual `denom`. Not sure if this will be a problem since we are checking for the original denom as oppose to the path and if the `tokenfactory` allows for modifying the denom. Will double check with the Osmosis team.